src/components/CustomerSignUp/CustomerSignUp.jsx

```diff
@@ -0,0 +1,205 @@
+import React, { useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import store from 'store';
+import PropTypes from 'prop-types';
+import { withRouter, Link } from 'react-router-dom';
+import Avatar from '@material-ui/core/Avatar';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
+import TextField from '@material-ui/core/TextField';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import Checkbox from '@material-ui/core/Checkbox';
+import Grid from '@material-ui/core/Grid';
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import useStyles from './Styles';
+import { CUSTOMER_CREATE, CUSTOMER_ACCESS_TOKEN_CREATE } from './GraphQl';
+import Spinner from '../Spinner';
+
+function CustomerSignUp({ history }) {
+  const classes = useStyles();
+  const [firstName, setFirstName] = useState(null);
+  const [lastName, setLastName] = useState(null);
+  const [email, setEmail] = useState(null);
+  const [password, setPassword] = useState(null);
+  const [phone, setPhone] = useState(null);
+  const [acceptsMarketing, setAcceptsMarketing] = useState(false);
+  const [formErrors, setFormErrors] = useState([]);
+
+  const [
+    customerAccessTokenCreate,
+    { loading: loadCreateToken, client }
+  ] = useMutation(CUSTOMER_ACCESS_TOKEN_CREATE, {
+    onCompleted: ({
+      customerAccessTokenCreate: { customerAccessToken, customerUserErrors }
+    }) => {
+      if (customerUserErrors.length > 0) {
+        setFormErrors(customerUserErrors);
+      } else {
+        client.writeData({
+          data: { customerToken: customerAccessToken.accessToken }
+        });
+        store.set('customer', {
+          token: customerAccessToken.accessToken,
+          expDate: customerAccessToken.expiresAt
+        });
+        history.push('/');
+      }
+    }
+  });
```

Comment:
I think it will be better for understanding if you will move this to some service or 'action'